### ISSUE TITLE
Prevent users redeem the output after the expiry

### ISSUE THREAD

**Issue by pavelivanov** [bitcoinjs/bitcoinjs-lib#1005]
Hi there. I can't understand how to write script to prevent users from redeeming the output AFTER the expiry. So I'd like to lock transactions after timeout. If there any way to do this?

**Comment by dcousens** [comment]
`OP_CHECKLOCKTIMEVERIFY` is enforced by cross-referencing the top-stack item against the transactions `nLockTime`,  if the top-stack item **is** greater,  the transaction is invalid.

What you want is to enforce that a transaction **has** a `nLockTime` lower than a particular block height... I don't know if that is possible.

**Comment by junderw** [comment]
Only way to do this currently is make a two branch OP_IF:

1. MULTISIG with YOU + USER
2. CHECKLOCKTIMEVERIFY + YOU ALONE

But you would need to set up a way for you to automatically sign the multisig for the user IF the timer is not over... then if the timer is over, refuse to sign the multisig.

**Comment by dcousens** [comment]
> then if the timer is over, refuse to sign the multisig.

tl;dr it may as well be solely redeemable by the user,  with the CLTV on `YOU` for after the timeout.  Then you simply *have* to sweep.

**Comment by junderw** [comment]
true.

remove multisig, and just retrieve ASAP the time limit is over.

**Comment by dcousens** [comment]
I don't think this is possible,  but I welcome any innovative ideas :grinning:
